dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,113 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(func, **func_kwargs):
+    """Split args on those applicable to Pipeline constructor and the decorated function."""
+    func_argspec = inspect.getfullargspec(func)
+    ctor_argspec = inspect.getfullargspec(Pipeline.__init__)
+
```

Comment:
```suggestion
    if func_argspec.varkw is not None:
        raise ValueError("Use of variadic keyword argument `**{}` in graph definition function is not allowed.".format(func_argspec.varkw))
```

Reply:
I agree that this is probably the simplest and cleanest solution, that we can easily document - just say that all the variadic kwargs are forwarded to the pipeline and the original function is not supposed to use it/have it.